types/node/index.d.ts

```diff
@@ -377,7 +383,7 @@ declare namespace NodeJS {
         abort(): void;
         chdir(directory: string): void;
         cwd(): string;
-        env: any;
+        env: { [key: string]: string | undefined };
```

Comment:
@andy-ms 

Instead of reverting is it possible to discuss the path forward? I am willing to make the required changes to make Travis CI happy (hopefully with some help).

In issue #16261 you mentioned that you think this should wait for the Node version 8 release. Node 8 has been released so maybe this is the time to proceed.

The reason why I am so keen about the correct definition of process.env, besides the invisibility for potential breakage with any, is the potential security risk involved. For example, the usage of secrets as environment variables, but no logic in the code when the environment variables with the secrets aren't available. Having the correct type for process.env can prevent such oversight.

cc-ing @mhegazy and @xeoneux, because you were also working on this in closed issue #16261